tune_sklearn/tune_basesearch.py

```diff
@@ -404,8 +404,14 @@ def _can_early_stop(self):
             bool: if the estimator can early stop
 
         """
-        return (hasattr(self.estimator, "partial_fit")
-                and callable(getattr(self.estimator, "partial_fit", None)))
+        sklearn_module = ".".join(
+            getattr(self.estimator, "__module__", None).split(".")[:-1])
+        return (hasattr(self.estimator, "partial_fit") and callable(
+            getattr(self.estimator, "partial_fit",
+                    None))) or (hasattr(self.estimator, "warm_start")
```

Comment:
The partial fit stuff is what we had before. This just adds an or condition to see if there's warm_start and max_iter. It also has to ensure that it isn't an ensemble or tree, because warm_start on these types of estimators doesn't allow you to continue fitting on the same estimator (you have to add more trees or more components in order to continue fitting, which isn't what we want to do).